File: 프로그래머스/unrated/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.py
def solution(board, k):
    m = len(board)
    n = len(board[0])
    total_sum = 0
    
    for i in range(m):
        for j in range(n):
            if i + j <= k:
                total_sum += board[i][j]
    
    return total_sum

File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int** board, size_t board_rows, size_t board_cols, int k) {
    int answer = 0;
    for(int r = 0; r < board_rows; r++) {
        for(int c = 0; c < board_cols; c++) {
            if(r + c > k) break;
            answer += board[r][c];
        }
    }
    return answer;
}